apps/shared/avifpng.c

```diff
@@ -448,6 +487,71 @@ avifBool avifPNGWrite(const char * outputFilename, const avifImage * avif, uint3
     if (avif->icc.data && (avif->icc.size > 0)) {
         png_set_iCCP(png, info, "libavif", 0, (png_iccp_datap)avif->icc.data, (png_uint_32)avif->icc.size);
     }
+
+    png_text texts[2];
+    int numTextMetadataChunks = 0;
+    if (avif->exif.data && (avif->exif.size > 0)) {
+#ifdef PNG_eXIf_SUPPORTED
+        if (avif->exif.size > UINT32_MAX) {
+            fprintf(stderr, "Error writing PNG: Exif metadata is too big\n");
+            goto cleanup;
+        }
+        png_set_eXIf_1(png, info, (png_uint_32)avif->exif.size, avif->exif.data);
+#else
+        if (!avifBytesToRawProfile(&avif->exif, "Exif", &exif)) {
+            fprintf(stderr, "Error writing PNG: Exif metadata is too big\n");
+            goto cleanup;
+        }
+        const png_text text = {
+            .compression = PNG_TEXT_COMPRESSION_zTXt,
+            .key = "Raw profile type exif",
+            .text = (char *)exif.data,
+            .text_length = exif.size,
```

Comment:
Older versions of Microsoft Visual Studio may not support a designated initializer. Unless we have tested this code with Visual Studio 2019 and 2017, it is best to replace this with a traditional initializer. Our CI Windows build uses Visual Studio 2022.

Reply:
From 2013 onward designated initializers should work with C99 code. The issues we've seen elsewhere are in C++ where they were only added in C++20; I'm not sure what level of support VS has for those.

Reply:
2019 seems to be the first to give a warning:
... error C7555: use of designated initializers requires at least '/std:c++20'